### PR TITLE
Implemented goto_line

### PR DESCRIPTION
**Pull request by k3m3**

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -83,6 +94,22 @@ impl Client {
         self.edit(view_id, "scroll", Some(json!([first_line, last_line])))
     }
 
+    pub fn goto_line(&mut self, view_id: ViewId, line: u64) -> ClientResult<()> {
+        self.edit(view_id, "goto_line", Some(json!({"line": line})))
+    }
+
+    pub fn copy(&mut self, view_id: ViewId) -> ClientResult<Value> {
+        self.ask(view_id, "copy", None as Option<Value>)
```
Comment (little-dude) [inline src/client.rs]: Sorry if this is a dumb question, but why do this one and `cut` need to be a requests while the others are notifications? I haven't looked at the API docs for a while, and after a quick look, it seems the doc don't specify whether a given method should be a notification or a request. A while ago, "edit" were always notifications I think.

File: src/client.rs [hunk 2]
```diff
@@ -59,6 +59,17 @@ impl Client {
         ))
     }
 
+    pub fn ask<T: Serialize>(
```
Comment (little-dude) [inline src/client.rs]: If we really need a request version of `edit`, maybe we should rename `edit` into `edit_notify` and this one into `edit_request`?

File: src/client.rs [hunk 3]
```diff
@@ -59,6 +59,17 @@ impl Client {
...
+        &mut self,
+        view_id: ViewId,
+        method: &str,
+        params: Option<T>)-> ClientResult<Value> {
+            match get_edit_params(view_id, method, params) {
+            Ok(value) => self.request("edit", value),
```
Comment (little-dude) [inline src/client.rs]: The formatting is a bit off here.

File: src/client.rs (hunk 1 above)
Comment (bytebuddha) [reply src/client.rs]: Copy And Cut are requests [xi docs](https://xi-editor.github.io/xi-editor/docs/frontend-protocol.html#copy)

File: src/client.rs (hunk 2 above)
Comment (bytebuddha) [reply src/client.rs]: I agree with having edit_notify & edit_request. I've actually meant to have done this a long time ago but the past few months have been busy at work :frowning_face:

File: src/client.rs [hunk 4]
```diff
@@ -13,7 +13,7 @@ pub type ClientResult<T> = Box<Future<Item = T, Error = ClientError> + Send>;
 #[derive(Clone)]
 pub struct Client(pub protocol::Client);
 
-fn get_edit_params<T: Serialize>(
+pub fn get_edit_params<T: Serialize>(
```
[the remark this answers is not in the document]
Comment (bytebuddha) [reply src/client.rs]: I would also prefer this not being exposed unless absolutely required

File: src/client.rs (hunk 1 above)
Comment (little-dude) [reply src/client.rs]: Oh thanks @bytebuddha, methods that return something are requests indeed. I shouldn't review PRs before my morning cup of coffee.

File: src/client.rs (hunk 2 above)
Comment (little-dude) [reply src/client.rs]: Cool. Do you mind making that change @k3m3 ?

File: src/client.rs (hunk 1 above)
Comment (bytebuddha) [reply src/client.rs]: No worries, were all guilty of that one sometimes. I'll be pretty quite the next couple of days but i'm good with merging this after the `ask` method is renamed if you want to go ahead and merge this after that

**Comment by k3m3** [comment]
Should be ok now, sry for the noise.

**Comment by little-dude** [comment]
@k3m3 do you mind renaming `edit` into `edit_notify`, as well?

**Pull request merged**